codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2PushPromiseFrameTest.java

```diff
@@ -90,8 +91,17 @@ protected void initChannel(SocketChannel ch) {
     }
 
     @Test
-    public void send() throws InterruptedException {
-        connectionFuture.sync();
+    public void send() {
+        connectionFuture.addListener(new ChannelFutureListener() {
+            @Override
+            public void operationComplete(ChannelFuture future) {
+                try {
+                    clientHandler.write();
```

Comment:
This method should not block... You need to not call `sync()` in it when you call it in a handler

Reply:
Done